### DISCUSSION TITLE
laravel 10 中无法使用withXX的简易写法

### DISCUSSION THREAD

**Discussion by wangtufly** [w7corp/easywechat#2741] [General]
$this->client->withMchId()->get('v3/pay/transactions/out-trade-no/'.$orderNumber);
这个withMchId()的简易写法是没用了么，get带不上mch_id;
`"message": "HTTP/1.1 400 Bad Request returned for \"https://api.mch.weixin.qq.com/v3/pay/transactions/out-trade-no/76369600681480192\".",
    "exception": "Symfony\\Component\\HttpClient\\Exception\\ClientException",
    "file": "/var/www/Help_player/vendor/symfony/http-client/Response/CommonResponseTrait.php",
    "line": 168,`

**Comment by wjfz** [comment]
这个接口不需要 mchid 啊，打印下输出看看。

另外就是`$this->client`这种写法很怪，正常情况下应该是
```
$app = new \EasyWeChat\Pay\Application($config = []);
$client = $app->getClient();
$client->get()
```

**Reply by wangtufly** [reply]
这个with的写法是文档上有点写法，正常调用是ok的，我用的是laravel的facade直接调用的

**Comment by wangtufly** [comment]
![image](https://github.com/w7corp/easywechat/assets/44032147/79834e83-6ef0-43b0-b526-85729b4475bb)
文档中的写法

**Comment by overtrue** [comment]
贴你的代码

**Reply by wangtufly** [reply]
![image](https://github.com/w7corp/easywechat/assets/44032147/0cfc9c89-ee40-48a0-8278-16887c403eb3)
zhe这段代码，本来想改成with带参数的形式，结果不行，出现上面的错误
本身引入的是  "overtrue/laravel-wechat": "^7.2",
构造函数中引入client：
 $this->app = EasyWeChat::pay();
 $this->client = $this->app->getClient();

**Reply by overtrue** [reply]
composer update

**Reply by wangtufly** [reply]
成功了成功了！！多谢大佬

**Comment by wangtufly** [comment]
> composer update
